app/app/static/custom/ourscript.js

```diff
@@ -4,85 +4,215 @@
-	let team = document.getElementById('team').value;
-	let enquiry = document.getElementById('enquiryType').value;
+	const studentName = document.getElementById('studentName').value;
+	const studentNumber = document.getElementById('studentNumber').value;
+	const unitCode = document.getElementById('unitCode').value;
+	const queue = document.getElementById('queue').value;
+	const enquiry = document.getElementById('enquiry').value;
 
-	
-	fetch("add_entry", {
-        method: "POST",
-        headers: {
-            studentName: name,
-            studentNumber: id,
-            unitCode: unit,
-            enquiry: enquiry,
-            queue: team
-        }
-    })
-
-	const table = $(`#${team == 'STUDYSmarter' ? 'SS' : 'lib'}QueueTable tbody`);
-	table.append(
-			`<tr id="${id}" class="initialTime">
-			<td>${name}</td>
-			<td>${id}</td>
-			<td>${unit}</td>
-			<td class="text-right">${enquiry}</td>
-			<td class="text-right"><label id="minutes${id}">00</label><label id="colon">:</label><label id="seconds${id}">00</label></td>
-			<td class="td-actions text-right">
-			<button type="button" rel="tooltip" class="btn btn-success" onclick="addSessionToTeam('${team}',${id})(this)"><i class="material-icons">how_to_reg</i></button>
-			<button type="button" rel="tooltip" class="btn btn-danger" onclick="deleteRow(this)"><i class="material-icons">close</i></button></td>
-			</tr>`
-	);
-
-	hideAddToQueue();
-	timers[id] = 0;
-	timerIntervals[id] = setInterval(setTime, 1000, id);
+	// Adds to state and rerender
+	addToQueueList({
+		studentName,
+		studentNumber,
+		unitCode,
+		queue,
+		enquiry
+	})
+
+	hideAddToQueueForm();
 });
 
-function deleteRow(x) {
-	$(x).parents('tr').remove();
+function deleteRow(id, status) {
+	const closureFunction = (currentElement) => {
+		console.log(currentElement)	
+		status = "Ended"
+		terminateSession({
+			id,
+			status
+		})
+}
+return closureFunction
 }
 
-function addSessionToTeam(team, id){
+function finishRow(id, status) {
+	const closureFunction = (currentElement) => {
+		console.log(currentElement)	
+		status = "Completed"
+		terminateSession({
+			id,
+			status
+		})
+}
+return closureFunction
+}
+
+function addSessionToTeam(id, status){
 	// This below is a function being stored to a variable that can be returned
 	const closureFunction = (currentElement) => {
-		console.log(currentElement)
-		var row = $(currentElement).parents('tr');
-		row.children().first().before(`<td>${team}</td>`);
-		row.children().last().remove();
-		row.children().last().after(`<td class="td-actions text-right"><button type="button" rel="tooltip" class="btn btn-success" onclick="deleteRow(this)"><i class="material-icons">how_to_reg</i></button></td>`);	
-		inSessionTable.append(row);
-		
-		clearInterval(timerIntervals[id]);
-		timers[id] = 0;
-		timerIntervals[id] = setInterval(setTime, 1000, id);
+		console.log(currentElement)	
+		status = "In Session"
+		clearInterval(id);		
```

Comment:
"id" is not an interval. You have to select an item from the timerInterval

Reply:
That clearInterval shouldn't be existed, deleted!